components/service/glean/src/main/java/mozilla/components/service/glean/storages/StringListsStorageEngine.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.storages
+
+import android.annotation.SuppressLint
+import mozilla.components.service.glean.CommonMetricData
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This singleton handles the in-memory storage logic for strings. It is meant to be used by
+ * the Specific Strings API and the ping assembling objects. No validation on the stored data
+ * is performed at this point: validation must be performed by the Specific Strings API.
+ *
+ * This class contains a reference to the Android application Context. While the IDE warns
+ * us that this could leak, the application context lives as long as the application and this
+ * object. For this reason, we should be safe to suppress the IDE warning.
+ */
+@SuppressLint("StaticFieldLeak")
+internal object StringListsStorageEngine : StringListsStorageEngineImplementation()
+
+internal open class StringListsStorageEngineImplementation(
+    override val logger: Logger = Logger("glean/StringsStorageEngine")
+) : GenericScalarStorageEngine<Array<String>?>() {
+
+    companion object {
+        // Maximum length of any list
+        private const val MAX_LIST_LENGTH_VALUE = 50
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
I'm getting an "Unchecked cast Any? to List<String>.  According to what I researched, this is normal as there is no way for it to do the generics type checking in this situation.  There were some solutions for manually checking each element of the list to ensure they were of String type, but they looked a little cumbersome.  If we are worried that this may try and deserialize a list that contains something other than a string, we could also just trap the exception that would be raised by the failed cast.  My choice was to assume we would only ever deserialize a list we had serialized and so I elected to suppress the warning.

Reply:
Certainly :)

Reply:
Let's remove the suppression here and change `return value as? List<String>` to `if (value is List<String>) value as List<String> else null`